packages/modules/pricing/integration-tests/__tests__/services/pricing-module/price-list.spec.ts

```diff
@@ -378,11 +381,14 @@ moduleIntegrationTestRunner<IPricingModuleService>({
               ends_at: "10/20/2030",
             },
           ])
-          expect(priceList).toEqual(
-            expect.objectContaining({
-              starts_at: new Date("10/10/2010").toISOString(),
-              ends_at: new Date("10/20/2030").toISOString(),
-            })
+
+          expect(priceList).toHaveProperty("starts_at")
+          expect(priceList).toHaveProperty("ends_at")
+          expect(priceList.starts_at?.toString()).toEqual(
+            new Date("10/10/2010").toString()
+          )
+          expect(priceList.ends_at?.toString()).toEqual(
+            new Date("10/20/2030").toString()
```

Comment:
For some weird reason, the field is a `string` after update. Maybe worth looking into?

Reply:
wouldnt it be that when it comes from the database it is always a string? 🤔 I think that's why originally we typed it as Date | string, because you can have either of those types depending if you are setting it or it comes from the DB. We can validate that if you want but I believe it is it

Reply:
as per our investigation, the place the serialization/deserialization occurs for the entity is here (node_modules/@mikro-orm/core/types/DateType.js) except that the update is relying on the upsertWithReplace which goes out of the classic path of managed entities and rather use native methods instead, which explain now why the behaviour is different between create and update